### ISSUE TITLE
Cannot provision ACI/Azure WebApp from ACR with private endpoint and disabled public access

### ISSUE THREAD

**Issue by bczudek** [Azure/acr#494]
**Describe the bug**
Following the guide: https://docs.microsoft.com/en-us/azure/container-registry/container-registry-private-link I have created Azure Container Registry with private endpoint (including private DNS zone) and disabled public access. I managed to login to it and pull/push images from the vm connected to the same virtual network however I couldn't create ACI or Azure WebApp on containers.

**To Reproduce**
1. Follow the guide https://docs.microsoft.com/en-us/azure/container-registry/container-registry-private-link to create ACR with private endpoint and disabled public access
2. Create a new ACI based on image stored in ACR configured in point 1 - it will fail stating that it can't get the image
3. Create a new WebApp based on image stored in ACR configured in point 1 - it will fail stating that it can't get the image
 
**Expected behavior**
Azure Container Instances or Azure Web App on Containers should be able to use the image from ACR which is configured with disabled public access 

are you aware of any workarounds?

Many thanks

**Comment by yugangw-msft** [comment]
It is a known issue that ACR and WebAPP is not able to access ACR behind the firewall, and they are working on it.
CC @dlepow we need to call out in the doc if have not

**Comment by dlepow** [comment]
@bczudek - This limitation is called out in a Note at the top of the article. Mentions ACI specifically (though not WebApp):

**Note**

Azure Security Center can't currently perform image vulnerability scanning in a registry that restricts access to private endpoints, selected subnets, or IP addresses. Instances of certain Azure services including Azure DevOps Services and Azure Container Instances are also unable to access a network-restricted container registry.

**Comment by dlepow** [comment]
Please close this issue as the limitation is currently documented. Thank you!

**Comment by grstreten** [comment]
> It is a known issue that ACR and WebAPP is not able to access ACR behind the firewall, and they are working on it. CC @dlepow we need to call out in the doc if have not

Is there a likely indication of when a future resolution could be expected, or somewhere this information could be found? Otherwise will keep an eye out. Thanks

**Comment by SteveLasker** [comment]
While I appreciate this is an Azure holistic limitation, this is a WebApp limitation, not specific to ACR. 
Adding @jeffhollan

**Comment by grstreten** [comment]
@SteveLasker makes sense, though I notice it also prevents DevOps from working properly. So does this fall into the web app 'basket'?

**Comment by jeffhollan** [comment]
And good news for web apps is this capability should be rolled out / is completing rollout now: https://azure.github.io/AppService/2021/07/03/Linux-container-from-ACR-with-private-endpoint.html

**Comment by SteveLasker** [comment]
It falls in the bucket of any service that doesn't yet support private vnets (private link).
If you put a resource in a private network and block external access, then is it the resources issue that others can't get access to it, because it's in a private vnet? :)

**Comment by ustrum** [comment]
Given the explicit mention to ACI being unable to contact ACR through private endpoint was removed from [the warning](https://github.com/MicrosoftDocs/azure-docs/blob/master/includes/container-registry-scanning-limitation.md), is it because it is expected to work now, or is there any expectancy on it to be working any time soon?

**Comment by dlepow** [comment]
#assign: @tejaswikolli-web

**Comment by dlepow** [comment]
#reassign: @tejaswikolli-web

**Comment by northtyphoon** [comment]
Please follow https://docs.microsoft.com/en-us/azure/container-instances/using-azure-container-registry-mi if you want to deploy images from ACR with private endpoint to ACI.
